### ISSUE TITLE
Large File Uploader (LFU) Electron App MVP

### ISSUE THREAD

**Issue by JeffLowe** [nftstorage/nft.storage#1015]
## Use case
User with very large file sizes
User with a large number of small files (10K+) in n-level deep directories
## Electron App MVP 
- [ ] No auth required. User generates API token in nft.storage account, and utilize this token in the Electron app. 
- [ ] Ability to upload anything that the JS client can upload (https://github.com/nftstorage/nft.storage/tree/main/packages/client).
- [ ] Upload UI interface that allows drag/drop selection of local directory.   (No folder observer needed, just a one-off selection and upload).
- [ ] Pre-upload table shows all directories and files targeted, with name, description, filename (with extension)
- [ ] Ability to change metadata name or description of individual files before upload
- [ ] No file type restrictions (for now)
- [ ] The app recursively walks upload directories “n” levels deep and uploads  
      - ** NOTE:  nft.storage doesnt preserve file names if file is at root of upload...see comment below for full and deeper context https://github.com/nftstorage/nft.storage/issues/1015#issuecomment-1004225093
- [ ] During upload, app should skip items previously uploaded (items already in the electron app’s local storage).
- [ ] Handles large files by running the DAG locally (the lib should do this)
- [ ] Post upload view, display everything that has been uploaded in a list in the app's localstorage, including at least name, description, CID, metadata
- [ ] Documentation / instructions draft on how to use this tool to do a 10K NFT drop
- [ ] Optional - Upload progress bar
- [ ] Optional - concurrent uploads (if there’s a gain)

**Comment by dchoi27** [comment]
Thanks! A few notes:

- NFT.Storage doesn't preserve file names if the file is at the root of the upload - only if the root is a directory
- For this reason, I don't think we want to upload files in a directory individually - the expectation from users is that file names are preserved if they upload a directory. For the 10K NFT drops, they like having the filenames shown in the URL. We do need a way to display individual files in the app though (e.g., they uploaded a directory with root CID `cid1` that contains 10 files, and the app shows 10 rows `cid1/file1`, `cid1/file2`, ...) - this differs from the Web UI which only would show a row for the root CID
- Given that large directories encounter problems when getting converted into CARs, we should split large directories up into smaller directories if they are too large (probably requires some testing to figure out the right threshold, but based on what we've heard in the wild, I think at least a 50MB directory should be handled fine)
- There should be a way to detect if an upload is interrupted and make this obvious to the user (even though per #896 the root CID will show up fine to the user)
- For `Documentation / instructions draft on how to use this tool to do a 10K NFT drop`, think we'll need a way to allow users to select and export a list of their files
- I know there was conversation on whether to show all uploads from the user in the app UI, or just the ones they've uploaded via the app. The answer is probably just the app - but sounds like we've made the decision to rely on local storage rather than make changes to the API/database to tag uploads via the app. Think that's fine, but want to flag that beyond the MVP, we probably then want a way for the user recover files from the service to the app if their computer dies, etc.

**Comment by JeffLowe** [comment]
The first 3 bullets I called out on the upload "individually" bullet in the description requirements (both @cmunns, @redaphid, and @dashcraft are aware). 

Interrupted upload alert I placed in a post-mvp ticket #1024 If it does need to be part of MVP, we can just switch it to be on the MVP epic ticket and we'll get it added. 

Ability for user to export text list of uploaded files, another post MVP feature ticketed here #1021

I believe that since they are using an api token from their nft.storage account, that any files uploaded via cli or app would show in their nft.storage /files page.  But, I'm clarifying this now.
